### DISCUSSION TITLE
How to create a new docs-like directory and menu.

### DISCUSSION THREAD

**Discussion by atwriter** [thuliteio/doks#173] [Q&A] [answered]
Hi all! Let me preface this that that is an awesome theme. Thank you so much for putting it together. I am NOT a web dev. I am quite noob at this so please be gentle :) . I have managed to get a site up and running and tweaked/added content using the doc and discussion topics here, but have run into a snare.

I have a need for a new docs archetype menu and folder structure similar to the doks public site:

![image](https://user-images.githubusercontent.com/54590260/110385064-16862480-8024-11eb-939f-4a11849aa126.png)

Tutorial is its own folder structure and sidebar menu. I am having a hard time reproducing that independent from the template /docs structure. What are the rough steps that must be done to have the menu structure work correctly with new folder?

So far, I have:

1. Created a new folder under /content.
2. Changed the menu options in menus.toml. I have the [[main]] link pointing to the new folder created. And added the new menu as well.

What else needs to happen generally?

Thank you!

**Comment by h-enk** [comment, marked as answer]
Hi @atwriter,

Thanks for your kind words — appreciated!

This is the general workflow:

1. Add pages to the new folder under /content — see also: [Add a documentation page](https://getdoks.org/tutorial/add-pages/#add-a-documentation-page)
2. Add first level menu items — see also: [Add to docs menu](https://getdoks.org/tutorial/add-menus/#add-to-docs-menu)
3. Add second level menu items in the frontmatter of a documentation page — see also: [Add to docs menu](https://getdoks.org/tutorial/add-menus/#add-to-docs-menu)

If you're still getting nowhere, please provide a link to your repository — then I'll do an initial setup for you.

**Reply by atwriter** [reply]
Hi @h-enk . Thanks so much for your reply. I appreciate it.

I followed your instructions and unfortunately am still running into the same problem. I successfully create a new content section but I can't get the sidebar menus to appear correctly in the new section and menu. 

I created a fresh clone of your template repo and am trying to build a new content section called "docs2." This section has its own menu and sidebar. I am still unable to create what I am attempting. The new section "docs2" does not display the sidebar.

Help would greatly appreciated.

https://github.com/atwriter/new_doks_site 

Thank you!

**Reply by h-enk** [reply]
Hi @atwriter, I created a PR: https://github.com/atwriter/new_doks_site/pull/1

Check the changes /see what I did. Then merge if you like.

Hope this helps!

**Reply by atwriter** [reply]
Eureka! Thank you very much! I was able to replicate what you did and it works as expected. :)

**Reply by roob811** [reply]
+1 For question and answer. With the commit in the https://github.com/atwriter/new_doks_site repo I could fix it as well! Thnx!

**Comment by FactoryOptic** [comment]
I am using this and it works great. Any chance we can get the search to work on the docs2 tab?

And is there any limit on adding docs3, doc4, docs5 etc?

Thank you

**Comment by h-enk** [comment]
First off, note, that you can also create a Docs based tree with a single command — see the Docs docs: [Docs based tree](https://getdoks.org/docs/overview/commands/#docs-based-tree)

Setting what to search (and what not) will be super easy with the upcoming Doks release (planned to be released end of next week) — see also https://github.com/h-enk/doks/pull/799. Until then, see https://github.com/h-enk/doks/discussions/395

The limit is determined by the size of the index file generated (I've tested with an index of 2000 pages and it looks workable — see also https://github.com/h-enk/doks/issues/738). Splitting the index would be a next step, but is not planned for.
